compat/mingw.c

```diff
@@ -560,6 +560,20 @@ int mingw_rmdir(const char *pathname)
 		return -1;
 
 	do {
+		/*
+		 * Contrary to Linux rmdir(), Windows' _wrmdir() and _rmdir()
+		 * will remove the directory at the path if it is a symbolic link
+		 * which leads to issues when symlinks are used in the .git folder
+		 * (in the context of Repo for instance). So before calling _wrmdir()
+		 * we first check if the path is a symbolic link. If it is, we exit
+		 * and return the same error as Linux rmdir() in this case (ENOTDIR).
+		 */
+		struct stat sb;
+		if (!mingw_lstat(pathname, &sb) && S_ISLNK(sb.st_mode)) {
+			errno = ENOTDIR;
+			break;
+		}
+
```

Comment:
Could you move this to before the loop? The loop retries to delete a directory in case it is still in use, and it does not really make sense to ask over and over again whether it is a symbolic link or not.

Also, regarding the commit message:

> issues especially when Repo is used.

The `repo` tool is not shipped with Git for Windows, and is actually not all that known outside of a small circle of Git users. It would make sense to elaborate a bit on what it is, and link to its homepage (if it has one).

Speaking of the commit message, I would love to see a paragraph added that talks about [`RemoveDirectoryW()`](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-removedirectoryw) having the same problem as `_wrmdir()`, and it is even documented:

> RemoveDirectory removes a directory junction, even if the contents of the target are not empty; the function removes directory junctions regardless of the state of the target object. For more information on junctions, see [Hard Links and Junctions](https://docs.microsoft.com/en-us/windows/desktop/FileIO/hard-links-and-junctions).

Well, _technically_ it does not talk about reparse points in general but only specifically about junctions. But the point is that we cannot expect that function to treat reparse points in the way we want, in general, and therefore we must take manual precautions (and there is no `RemoveDirectoryExW()` that would hypothetically let the caller specify whether it is okay to delete reparse points).

Since that documentation also talks about hard links, I briefly investigated whether `_wrmdir()` could potentially remove hardlinks by mistake. But hardlinks can only link to files, not directories, and `_wrmdir()` correctly errors out upon trying to remove hardlinks.

Reply:
I moved the code and I updated the commit message based on your recommendations.
Please let me know if it is what you had in mind.

Reply:
I like the new commit message!